### ISSUE TITLE
T3C: Dataset Plan - Capture, Labeling & Augmentation 🏷️

### ISSUE THREAD

**Issue by vikraman2212** [vikraman2212/sports-analyst#23]
# Task 3C: Dataset Plan - Capture, Labeling & Augmentation

**Status:** 📋 PLANNED  
**Parent:** #4 (T3: YOLOv8 Migration)  
**Duration:** 3 days  
**Start Date:** 2025-11-02  
**Owner:** ML  
**Dependencies:** T3B (#22)

---

## 🎯 Objective

Define dataset collection, labeling workflow, and augmentation strategy for training a custom YOLOv8 model on cricket ball detection.

**Note:** Only proceed with this task if T3B benchmark shows pretrained models are insufficient (recall < 0.85 or FPS < 20).

## 📦 Deliverables

- [ ] **Data Collection Plan**
  - Target: 500+ labeled images minimum
  - Capture guidelines (lighting, angles, backgrounds)
  - Diversity requirements (ball types, conditions)
  
- [ ] **Labeling Workflow & Guidelines**
  - Annotation tool selection (Roboflow, LabelImg, CVAT)
  - Bounding box guidelines (tight vs. loose)
  - Class definitions (cricket ball, other objects)
  - Quality control process
  
- [ ] **Small-Object Augmentation Strategy**
  - Augmentation techniques (mosaic, copy-paste, scale)
  - Rationale for cricket ball (small, fast-moving)
  - Configuration for YOLOv8 training

- [ ] **Dataset YAML Scaffold**
  - Path configuration
  - Train/val/test splits (70/20/10)
  - Class names and IDs

## ✅ Acceptance Criteria

- [ ] Labeling guideline doc created (`docs/T3C_labeling_guidelines.md`)
- [ ] Dataset YAML scaffold prepared (`ml/dataset/cricket_ball.yaml`)
- [ ] If proceeding: ≥500 labeled images collected and validated
- [ ] Augmentation config defined in training plan

### Minimum Dataset Size
- **Training Set:** 350+ images
- **Validation Set:** 100+ images
- **Test Set:** 50+ images
- **Total:** 500+ labeled images

### Diversity Requirements

**Lighting Conditions:**
- Bright sunny (40%)
- Overcast/cloudy (30%)
- Indoor artificial (20%)
- Twilight/low light (10%)

**Ball Types:**
- Red cricket ball (50%)
- White cricket ball (30%)
- Pink cricket ball (10%)
- Youth/practice ball (10%)

**Backgrounds:**
- Grass pitch (50%)
- Concrete/indoor (30%)
- Matting pitch (20%)

**Angles:**
- Side-on (bowler's perspective) - 60%
- Behind stumps - 20%
- 45° angle - 20%

### Annotation Tool
**Recommended:** Roboflow (free tier + export to YOLO format)
- **Alternative:** LabelImg, CVAT, Label Studio

### Bounding Box Rules
1. **Tight Boxing:** Box tightly around visible ball pixels
2. **Motion Blur:** Include blur within box (represents ball extent)
3. **Partial Occlusion:** Label if >50% of ball visible
4. **Multiple Balls:** Label all balls in frame (practice scenarios)

### Class Schema
```yaml
classes:
  0: cricket_ball  # Primary class
  # Optional (future):
  # 1: bat
  # 2: stumps
```

### Quality Control
- **Review:** 10% of annotations reviewed by second labeler
- **Consistency Check:** Random sample validation every 100 images
- **Rejection Criteria:** Incorrect class, loose box, missed ball

### Why Small-Object Augmentation?
Cricket balls are **small objects** in frame (typically <5% of image area), moving at high speed. Standard augmentation may not be sufficient.

### Recommended Techniques

**1. Mosaic Augmentation** (YOLO default)
- Combine 4 images into one
- Increases small object instances per batch

**2. Copy-Paste Augmentation**
- Copy ball from one frame, paste into another
- Increases ball diversity and count

**3. Scale Augmentation**
- Random scaling (0.5× to 1.5×)
- Helps model learn balls at various distances

**4. Small-Object Focus**
```yaml
# YOLOv8 config overrides
mosaic: 1.0         # Always use mosaic
copy_paste: 0.3     # 30% probability
scale: 0.5          # Scaling factor
hsv_h: 0.015        # Hue variation (minimal)
hsv_s: 0.7          # Saturation variation
hsv_v: 0.4          # Value/brightness variation
```

### Augmentation Pipeline
1. **Geometric:** Flip, rotate (small angles), scale
2. **Color:** HSV adjustments (lighting variation)
3. **Mosaic:** Combine images for dense training
4. **Copy-Paste:** Add more ball instances

## 📁 Output Files

Create in `ml/dataset/`:
- `cricket_ball.yaml` - Dataset configuration
- `images/train/` - Training images
- `images/val/` - Validation images
- `images/test/` - Test images
- `labels/train/` - YOLO format annotations
- `labels/val/`
- `labels/test/`

Create in `docs/`:
- `T3C_labeling_guidelines.md` - Full labeling instructions
- `T3C_dataset_plan.md` - Data collection strategy
- `T3C_augmentation_config.md` - Augmentation rationale

## 🔧 Dataset YAML Template

```yaml
# cricket_ball.yaml
path: /path/to/dataset
train: images/train
val: images/val
test: images/test

names:
  0: cricket_ball

# Training config (used in T3D)
augmentation:
  mosaic: 1.0
  copy_paste: 0.3
  scale: 0.5
  hsv_h: 0.015
  hsv_s: 0.7
  hsv_v: 0.4
```

## 🔗 Dependencies

**Upstream:** T3B (Benchmark) - only proceed if pretrained insufficient  
**Downstream:** T3D (Training) - uses this dataset

## 📚 References

- Roboflow Annotation: https://docs.roboflow.com/annotate
- YOLOv8 Data Format: https://docs.ultralytics.com/datasets/detect/
- Small Object Detection: https://docs.ultralytics.com/guides/small-objects/
- Augmentation Guide: https://docs.ultralytics.com/modes/train/#augmentation-settings

---

**Epic:** #4 (T3)  
**Previous:** T3B (Benchmark)  
**Next:** T3D (Training)  
**Conditional:** Only if T3B shows pretrained models inadequate
